api/user/schema.py

```diff
@@ -159,6 +159,14 @@ def mutate(self, info, email, **kwargs):
         return DeleteUser(user=exact_query_user)
 
 
+def restrictingAdmin(new_role):
```

Comment:
@cop1fab please add a docstring to give an overview of how this function works. Ensure to highlight the parameters it takes in and what it returns.